### ISSUE TITLE
ELECTRON_ASAR.js:158 Uncaught Error: The specified module could not be found. \\?\C:\Users\HzPC\AppData\Local\Temp\937.tmp.node

### ISSUE THREAD

**Issue by codehz** [electron-userland/electron-builder#580]
* **Version**: 5.12.1

* **Target**: Windows x64

ELECTRON_ASAR.js:158 Uncaught Error: The specified module could not be found. \\?\C:\Users\HzPC\AppData\Local\Temp\937.tmp.node

It seems `\\?\` shouldn't appear...

**Comment by develar** [comment]
Runtime error or build error? Please attach build log.

**Comment by codehz** [comment]
runtime error

**Comment by develar** [comment]
Please attach build log.

**Comment by codehz** [comment]
This is a problem I use the asar...And I found the answer in https://github.com/electron-userland/electron-packager/issues/424#issuecomment-232313870

**Comment by dpalou** [comment]
Hi @codehz,

how did you fix this?

I have the same problem, so I tried using this option in the package.json file:

"asarUnpack": "**/*.node"

And now I'm getting this error:

```
Error: The specified module could not be found.
\\?\C:\Program Files\WindowsApps\3312ADB7.MyApp_3.3.0.0_x64__t8q4t8fsbshw4\app\resources\app.asar.unpacked\node_modules\@nodert-win10\windows.data.xml.dom\build\Release\binding.node
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:173:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:187:18)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\Program Files\WindowsApps\3312ADB7.MyApp_3.3.0.0_x64__t8q4t8fsbshw4\app\resources\app.asar\node_modules\@nodert-win10\windows.data.xml.dom\lib\main.js:19:20)
    at Object.<anonymous> (C:\Program Files\WindowsApps\3312ADB7.MyApp_3.3.0.0_x64__t8q4t8fsbshw4\app\resources\app.asar\node_modules\@nodert-win10\windows.data.xml.dom\lib\main.js:64:3)
```

I've checked it and the _binding.node_ file exists in that path. Any idea on how to fix this?

 I also tried using no asar at all (asar: false), but the result is the same.

Thanks!

**Comment by codehz** [comment]
@dpalou https://github.com/electron/asar/issues/75

**Comment by revati-kiwi** [comment]
hi @dpalou 

did you find the fix to the issue?

Thanks!

**Comment by burakkp** [comment]
Hey Folks, 

I fixed this issue from my project. I removed --asar from in my package.json.

**Comment by mshenawy** [comment]
> Hey Folks,
> 
> I fixed this issue from my project. I removed --asar from in my package.json.

thanks, this worked with me

**Comment by danielehrhardt** [comment]
> Hey Folks,
> 
> I fixed this issue from my project. I removed --asar from in my package.json.

if i disable asar i get:
`asar usage is disabled — this is strongly not recommended  solution=enable asar and use asarUnpack to unpack files that must be externally available`

is there any other solution to this Problem?

**Comment by Esinko** [comment]
Hello everyone. I encountered this issue myself a few days ago and I've figured out how to fix it.
Basically, make sure **any** native files are not packed (not in the `.asar` file and instead in the asarUnpacked directory).
This can be achieved with Electron-Forge (for example) by adding the following to your packager config:

```json
"asar": {
    "unpack": "**/+(*.node|*.dll|)"
}
```

The idea is to make sure any native addons (`.node` files) or other files used by them are accessible natively from the filesystem.

That's all. I hope this helps one of you!
